internal/services/compute/virtual_machine_implicit_data_disk_resource.go

```diff
@@ -0,0 +1,462 @@
+package compute
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/compute/2022-03-02/snapshots"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/compute/2024-03-01/virtualmachines"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/locks"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/compute/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/compute/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var (
+	_ sdk.Resource                   = VirtualMachineImplicitDataDiskResource{}
+	_ sdk.ResourceWithUpdate         = VirtualMachineImplicitDataDiskResource{}
+	_ sdk.ResourceWithCustomImporter = VirtualMachineImplicitDataDiskResource{}
+)
+
+type VirtualMachineImplicitDataDiskResource struct{}
```

Comment:
Per the PR description:

> this new resource creates a new data disk when updating a VM and allows the system to place the this disk closer to the parent VM to provide better latency and align the fault domain of the disk with the parent VM for providing better reliability.

Can you provide a source showing the performance benefits of this? Last I read using a Zone/Proximity Placement Group should give the same benefit here - and from what I understood was replacing the Fault Domains concept (which is associated with Availability Sets, which I believe aren't recommended anymore?)?

Based on this issue (https://github.com/Azure/azure-cli/issues/28686) - since this exists only for copies of a Disk from a Snapshot, rather than a net-new Disk - I'm concerned that the resource name is going to be misleading too?